core/lib/dal/src/models/storage_eth_tx.rs

```diff
@@ -22,6 +22,10 @@ pub struct StorageEthTx {
     pub updated_at: NaiveDateTime,
     // TODO (SMA-1614): remove the field
     pub sent_at_block: Option<i32>,
+    // A `EIP_4844_TX_TYPE` transaction blob sidecar.
+    //
+    // Format a `bincode`-encoded `EthTxBlobSidecar` enum.
+    pub blob_sidecar: Option<Vec<u8>>,
```

Comment:
Maybe a stupid question, but still. If I'm not mistaken, blobs are taken as whole, even if you use a fraction of it. So we may assume that at times, the blob won't be fully utilized (e.g. each time when pubdata wasn't the criterion to seal the batch), and there will be an "empty" reminder.

Do we want to use any kind of compression when we store/load data? Given that the state size is a problem already, probably we would like to optimize on this front whenever possible.

Reply:
yes well my current idea here is that as you see `Vec<u8>` is used for all fields while from the protocol level we need three arrays: `[u8; 32]` `[u8; 48]` and `[u8; 32*4096]`. Two reasons for this:
1. `serde` does not implement traits for such const-size arrays
2. we actually have to store actual data with variable lengths for blob instead of a const array.

So as such there will always be a layer of conversion between a blob expanded to const size and `Vec<u8>` that is stored in DB and back. We just haven't figured out the ergonomics with @koloz193 but that is the current idea and things will remain as they are now. If we should use some compression for it is another topic; however the versioned blob struct should make it possible to do so in the future if necessary.

Reply:
we can definitely compress the data at the db level. there are a few things that happen before we store the blob:
1. we take the raw pubdata and split it into chunks of size `4096*31` lets call these `zksync_blobs`
2. for each `zksync_blob` we then convert it into an `ethereum_blob`
3. from the `ethereum_blob` we compute the other kzg values needed

while the `zksync_blob` might have trailing zeroes, im not sure if/how many trailing zeroes are in the `ethereum_blob`. for this we'd want to check if we want to store the `zksync_blob` or the `ethereum_blob`. this would introduce a crypto dependency to be able to convert `zksync_blob` -> `ethereum_blob`